File: Proyecto_Veterinaria_BlueJ/Cliente.java
import java.util.List;
import java.util.ArrayList;

public class Cliente {
    private String nombre;
    private String telefono;
    private List<Mascota> mascotas;

    public Cliente(String nombre, String telefono) {
        this.nombre = nombre;
        this.telefono = telefono;
        this.mascotas = new ArrayList<>();
    }

    public void registrarMascota(Mascota m) {
        this.mascotas.add(m);
    }
}

File: Control_Clinico_Veterinario/Control_Clinico_Veterinario/Cliente.cs
using System;
using System.Collections.Generic;

namespace SistemaVeterinaria
{
    public class Cliente : Persona
    {
        public List<Mascota> Mascotas { get; set; }

        public Cliente(string nombre, string telefono) : base(nombre, telefono)
        {
            Mascotas = new List<Mascota>();
        }

        public void RegistrarMascota(Mascota mascota)
        {
            Mascotas.Add(mascota);
            Console.WriteLine($"Mascota {mascota.Nombre} registrada al tutor {Nombre}.");
        }

        // Implementación polimórfica
        public override void MostrarDetalles()
        {
            Console.WriteLine($"[CLIENTE] {Nombre} | Tel: {Telefono} | Mascotas registradas: {Mascotas.Count}");
        }
    }
}